contracts/erc7683/ERC7683OrderDepositor.sol

```diff
@@ -244,7 +244,7 @@ abstract contract ERC7683OrderDepositor is IOriginSettler {
         relayData.message = acrossOrderData.message;
         fillInstructions[0] = FillInstruction({
             destinationChainId: SafeCast.toUint64(acrossOrderData.destinationChainId),
-            destinationSettler: _destinationSettler(acrossOrderData.destinationChainId).toBytes32(),
+            destinationSettler: acrossOrderData.destinationSettler.toBytes32(),
```

Comment:
@mrice32 how do we enforce the correct destinationSettler is called with [fill()](https://github.com/across-protocol/contracts/blob/466127e11673fb662e6c2cf3a8436c1606269ed3/contracts/SpokePool.sol#L1131)? The fill function doesn't seem to check destinationSettler at all.

Reply:
here's my workaround for now: 87261babad57945fde7207b6b5efbaea3ba4c737

This should prevent the filler from calling fill() on the incorrect destinationSettler.

Reply:
Hmmmm, this raises an interesting point and may make this strategy not work.

If the user specifies a malicious destination settler contract, the filler could be exploited. Is it the origin settler's responsibility to validate that if the filler correctly follows the fill instructions that they will be repaid? I think yes, but if we allow the order to specify an arbitrary destination settler that is unverifiable, then it's impossible to do that verification...

Thoughts?

Reply:
I do think the filler shouldn't blindly submit calldata on the destination chain, right? They should probably be aware of known destination settlers, and then only fill deposits that target those.